runtime/common/src/chain_ext.rs

```diff
@@ -174,3 +180,49 @@ pub fn decode<T: Decode>(input: Vec<u8>) -> Result<T, codec::Error> {
 	let mut input = input.as_slice();
 	T::decode(&mut input)
 }
+
+impl ChainExtensionError {
+    pub fn as_u32(&self) -> u32 {
+        match self {
+            ChainExtensionError::Other => 1,
```

Comment:
just curious: any reason why the number doesn't start at 0?

Reply:
Yeah, the index of this error will be returned to the calling contract with [RetVal::Converging(index)](https://crates.parity.io/pallet_contracts/chain_extension/enum.RetVal.html). And a `0` would generally indicate a success. That's why it makes sense that the errors start at `1`.

Reply:
Yeah we should definitely call it ChainExtensionOutcome and start at 0 with success. Otherwise it's too confusing.